camayoc/tests/qcs/cli/test_sources.py

```diff
@@ -0,0 +1,879 @@
+
+    qpc_source_edit = pexpect.spawn(
+        'qpc source edit --name {} --port {}'
+        .format(invalid_name, new_port)
+    )
+    qpc_source_edit.logfile = BytesIO()
+    assert qpc_source_edit.expect(
+        'Source "{}" does not exist.'.format(invalid_name)) == 0
+    assert qpc_source_edit.expect(pexpect.EOF) == 0
+    qpc_source_edit.close()
+    assert qpc_source_edit.exitstatus == 1
+
+
+def test_clear(isolated_filesystem, qpc_server_config):
+    """Clear a source.
+
+    :id: 45ec3f8a-b554-4578-886c-f1cb5ab42ffa
+    :description: Clear a source entry by entering the ``--name`` of an
+        already created entry.
+    :steps: Run ``qpc source clear --name <name>``
+    :expectedresults: The source entry is removed.
+    """
+    cred_name = utils.uuid4()
+    name = utils.uuid4()
+    hosts = '127.0.0.1'
+    cred_add(
+        {
+            'name': cred_name,
+            'username': utils.uuid4(),
+            'password': None,
+        },
+        [(CONNECTION_PASSWORD_INPUT, utils.uuid4())],
+    )
+
+    qpc_source_add = pexpect.spawn(
+        'qpc source add --name {} --cred {} --hosts {} --type network'
+        .format(name, cred_name, hosts)
+    )
+    assert qpc_source_add.expect('Source "{}" was added'.format(name)) == 0
+    assert qpc_source_add.expect(pexpect.EOF) == 0
+    qpc_source_add.close()
+    assert qpc_source_add.exitstatus == 0
+
+    source_show(
+        {'name': name},
+        '{{\r\n'
+        '    "credentials": \[\r\n'
+        '        {{\r\n'
+        '            "id": \\d+,\r\n'
+        '            "name": "{}"\r\n'
+        '        }}\r\n'
+        '    \],\r\n'
+        '    "hosts": \[\r\n'
+        '        "{}"\r\n'
+        '    \],\r\n'
+        '    "id": \\d+,\r\n'
+        '    "name": "{}",\r\n'
+        '    "port": 22,\r\n'
+        '    "source_type": "network"\r\n'
+        '}}\r\n'
+        .format(cred_name, hosts, name)
+    )
+
+    qpc_source_clear = pexpect.spawn(
+        'qpc source clear --name={}'.format(name)
+    )
+    assert qpc_source_clear.expect(
+        'Source "{}" was removed'.format(name)
+    ) == 0
+    assert qpc_source_clear.expect(pexpect.EOF) == 0
+    qpc_source_clear.close()
+    assert qpc_source_clear.exitstatus == 0
+
+    qpc_source_clear = pexpect.spawn(
+        'qpc source clear --name={}'.format(name)
+    )
+    assert qpc_source_clear.expect(
+        'Source "{}" was not found'.format(name)
+    ) == 0
+    assert qpc_source_clear.expect(pexpect.EOF) == 0
+    qpc_source_clear.close()
+    assert qpc_source_clear.exitstatus == 1
+
+    qpc_source_show = pexpect.spawn(
+        'qpc source show --name={}'.format(name)
+    )
+    assert qpc_source_show.expect(
+        'Source "{}" does not exist.'.format(name)
+    ) == 0
+    assert qpc_source_show.expect(pexpect.EOF) == 0
+    qpc_source_show.close()
+
+
+def test_clear_negative(isolated_filesystem, qpc_server_config):
+    """Clear a source which is not created.
+
```

Comment:
Suggest "Attempt to clear a source that does not exist"

Reply:
Much better, thank you